File: posawesome/posawesome/api/pos_profile.py
# Copyright (c) 2026, Youssef Restom and contributors
# For license information, please see license.txt

import frappe
from frappe import _


def validate(doc, method):
    validate_additional_price_lists(doc)


def validate_additional_price_lists(doc):
    """Ensure the extra price lists configured for the POS item list are usable.

    Each row must reference a Price List other than the POS Profile's own
    default (``selling_price_list``) and no Price List may be added twice,
    otherwise the POS item list would show a duplicate or redundant column.
    """

    rows = doc.get("posa_additional_price_lists") or []
    if not rows:
        return

    seen = set()
    for row in rows:
        price_list = row.price_list
        if not price_list:
            continue
        if doc.selling_price_list and price_list == doc.selling_price_list:
            frappe.throw(
                _("Row #{0}: {1} is already the default Price List for this POS Profile.").format(
                    row.idx, price_list
                )
            )
        if price_list in seen:
            frappe.throw(
                _("Row #{0}: {1} has already been added to the additional price lists.").format(
                    row.idx, price_list
                )
            )
        seen.add(price_list)

File: posawesome/posawesome/api/pos_profile.js
// Copyright (c) 20201 Youssef Restom and contributors
// For license information, please see license.txt

frappe.ui.form.on("POS Profile", {
	setup: function (frm) {
		frm.set_query("posa_cash_mode_of_payment", function (doc) {
			return {
				filters: { type: "Cash" },
			};
		});

		// Prevent selecting the POS Profile's own default price list (or a
		// price list already added) as an additional price list column.
		frm.set_query("price_list", "posa_additional_price_lists", function (doc) {
			const already_selected = (doc.posa_additional_price_lists || [])
				.map((row) => row.price_list)
				.filter(Boolean);
			const excluded = doc.selling_price_list
				? [doc.selling_price_list, ...already_selected]
				: already_selected;
			if (!excluded.length) {
				return {};
			}
			return {
				filters: [["Price List", "name", "not in", excluded]],
			};
		});

		frappe.call({
			method: "posawesome.posawesome.api.utilities.get_language_options",
			callback: function (r) {
				if (!r.exc) {
					frm.fields_dict["posa_language"].df.options = r.message;
					frm.refresh_field("posa_language");
				}
			},
		});
	},
});
